src/runtime/storageNG/store.ts

```diff
@@ -91,26 +132,114 @@ export class DirectStore<T extends CRDTTypeRecord> extends ActiveStore<T> {
   }
   // The driver will invoke this method when it has an updated remote model
   async onReceive(model: T['data'], version: number): Promise<void> {
-    const {modelChange, otherChange} = this.localModel.merge(model);
-    await this.processModelChange(modelChange, otherChange, version, true);
+    this.pendingDriverModels.push({model, version});
+    if (this.state === DirectStoreState.AwaitingResponse || this.state === DirectStoreState.AwaitingResponseDirty) {
+      return;
+    }
+    this.applyPendingDriverModels();
   }
   
-  private async processModelChange(thisChange: CRDTChange<T>, otherChange: CRDTChange<T>, version: number, messageFromDriver: boolean) {
+  private deliverCallbacks(thisChange: CRDTChange<T>) {
     if (thisChange.changeType === ChangeType.Operations && thisChange.operations.length > 0) {
       this.callbacks.forEach((cb, id) => cb({type: ProxyMessageType.Operations, operations: thisChange.operations, id}));
     } else if (thisChange.changeType === ChangeType.Model) {
       this.callbacks.forEach((cb, id) => cb({type: ProxyMessageType.ModelUpdate, model: thisChange.modelPostChange, id}));
     }
-       
+  }
+
+  private async processModelChange(modelChange: CRDTChange<T>, otherChange: CRDTChange<T>, version: number, fromDriver: boolean) {
```

Comment:
This is now only ever called from `onProxyMessage` with `fromDriver` always set to false.

Reply:
ah, thanks! Removed the extra parameter.